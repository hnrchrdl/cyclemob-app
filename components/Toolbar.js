import React from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import PropTypes from 'prop-types';

Toolbar.propTypes = {
  onToggleShowSearch: PropTypes.func.isRequired,
  onToggleUserLocation: PropTypes.func.isRequired,
  onToggleShowMenu: PropTypes.func.isRequired,
  followUserLocation: PropTypes.bool.isRequired
};

function Toolbar({
  onToggleShowSearch,
  onToggleUserLocation,
  followUserLocation,
  onToggleShowMenu
}) {
  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.toolbarItem} onPress={onToggleShowSearch}>
        <MaterialIcons name="search" size={20} />
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.toolbarItem}
        onPress={onToggleUserLocation}
      >
        <MaterialIcons
          style={followUserLocation ? { color: 'tomato' } : {}}
          name="center-focus-strong"
          size={20}
        />
      </TouchableOpacity>
      <TouchableOpacity style={styles.toolbarItem} onPress={onToggleShowMenu}>
        <MaterialIcons name="menu" size={20} />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    height: 50,
    backgroundColor: '#fff',
    opacity: 0.9
  },
  toolbarItem: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent'
  }
});
export default Toolbar;
