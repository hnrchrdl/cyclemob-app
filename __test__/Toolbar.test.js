import React from 'react';
import Toolbar from '../components/Toolbar';

import renderer from 'react-test-renderer';

const props = {
  onToggleShowSearch: () => {},
  onToggleUserLocation: () => {},
  onToggleShowBikecomputer: () => {},
  onToggleShowMenu: () => {},
  followUserLocation: false
};
it('renders without crashing', () => {
  const rendered = renderer.create(<Toolbar {...props} />).toJSON();
  expect(rendered).toBeTruthy();
});
