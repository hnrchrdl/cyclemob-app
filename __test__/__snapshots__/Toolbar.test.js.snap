// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toolbar
    followUserLocation={false}
    onToggleShowBikecomputer={[MockFunction]}
    onToggleShowMenu={[MockFunction]}
    onToggleShowSearch={[MockFunction]}
    onToggleUserLocation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="search"
            size={20}
          />
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="center-focus-strong"
            size={20}
            style={Object {}}
          />
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="info-outline"
            size={20}
          />
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="menu"
            size={20}
          />
        </TouchableOpacity>,
      ],
      "style": Object {
        "backgroundColor": "rgba(255,255,255,.95)",
        "flex": 1,
        "flexDirection": "row",
        "height": 50,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Icon
            allowFontScaling={false}
            name="search"
            size={20}
          />,
          "onPress": [MockFunction],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "name": "search",
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Icon
            allowFontScaling={false}
            name="center-focus-strong"
            size={20}
            style={Object {}}
          />,
          "onPress": [MockFunction],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "name": "center-focus-strong",
            "size": 20,
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Icon
            allowFontScaling={false}
            name="info-outline"
            size={20}
          />,
          "onPress": [MockFunction],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "name": "info-outline",
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Icon
            allowFontScaling={false}
            name="menu"
            size={20}
          />,
          "onPress": [MockFunction],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "name": "menu",
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="search"
              size={20}
            />
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="center-focus-strong"
              size={20}
              style={Object {}}
            />
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="info-outline"
              size={20}
            />
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="menu"
              size={20}
            />
          </TouchableOpacity>,
        ],
        "style": Object {
          "backgroundColor": "rgba(255,255,255,.95)",
          "flex": 1,
          "flexDirection": "row",
          "height": 50,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Icon
              allowFontScaling={false}
              name="search"
              size={20}
            />,
            "onPress": [MockFunction],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "name": "search",
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Icon
              allowFontScaling={false}
              name="center-focus-strong"
              size={20}
              style={Object {}}
            />,
            "onPress": [MockFunction],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "name": "center-focus-strong",
              "size": 20,
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Icon
              allowFontScaling={false}
              name="info-outline"
              size={20}
            />,
            "onPress": [MockFunction],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "name": "info-outline",
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Icon
              allowFontScaling={false}
              name="menu"
              size={20}
            />,
            "onPress": [MockFunction],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "name": "menu",
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
